### PR TITLE
Use absolute path for CompilerGeneratedFilesOutputPath

### PR DESCRIPTION
**Pull request by jjonescz**
Workaround for https://github.com/dotnet/roslyn/issues/75680 until the fix flows into VS IntPreview.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
     <!-- Enable output of generated files back into our Generated directory -->
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
-    <CompilerGeneratedFilesOutputPath>Generated</CompilerGeneratedFilesOutputPath>
+    <CompilerGeneratedFilesOutputPath>$(MSBuildThisFileDirectory)\Generated</CompilerGeneratedFilesOutputPath>
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj:20]: Can we add a note to the issue to undo this change and put a link to the issue here?

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj (hunk 1 above)
Comment (jjonescz) [reply src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj:20]: We don't necessarily need to undo it, absolute path should work always

Comment (chsienki) [reply src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj:20]: Yeah, I'm fine with leaving it as absolute as in the project file it means the same thing ultimately.

**Review by chsienki — APPROVED** [review]
